src/tensorwaves/optimizer/callbacks.py

```diff
@@ -84,6 +85,48 @@ def finalize(self) -> None:
         self.__stream.close()
 
 
+class CSVSummary(Callback):
+    def __init__(
+        self,
+        filename: str,
+        estimator: Estimator,
+        step_size: int = 10,
+    ) -> None:
+        """Log fit parameters and the estimator value to a CSV file."""
+        self.__function_call = 0
+        self.__step_size = step_size
+        self.__stream = open(filename, "w")
+        _empty_file(self.__stream)
+        if not isinstance(estimator, Estimator):
+            raise TypeError(f"Requires an in {Estimator.__name__} instance")
+        self.__estimator_type: str = estimator.__class__.__name__
+
+    def __call__(self, parameters: dict, estimator_value: float) -> None:
+        self.__function_call += 1
+        if self.__function_call % self.__step_size != 0:
+            return
+        output_dict = {
+            "Time": datetime.now(),
+            "Iteration": self.__function_call,
+            "Estimator_Type": self.__estimator_type,
+            "Estimator_Value": float(estimator_value),
+            # name: float(value) for name, value in parameters.items()
+        }
+        for name, value in parameters.items():
+            output_dict[name] = float(value)
+
+        df = DataFrame(output_dict, index=[self.__function_call])
+        df.to_csv(
+            self.__stream,
+            mode="a",
+            header=self.__function_call == 10,
```

Comment:
This looks a bit dirty. It would be best to do the initialization in the init of the Callback.

Reply:
This is how it's done in the YAMLWriter
https://github.com/ComPWA/tensorwaves/blob/d5dedc77fc16ca41da3f31a47280e8e7fb5c178f/src%2Ftensorwaves%2Foptimizer%2Fcallbacks.py#L155-L157

Reply:
Oh sorry nvm